### PR TITLE
Initial Project

### PR DESCRIPTION
**Pull request by TrustHenry**
This is the initial configuration of the stoa API project,
which operates in a Node.js environment and is implemented with TypeScript.

### CODE REVIEW

**Comment by Geod24** [comment]
Please also add a CI ;)

**Comment by Geod24** [comment]
https://github.com/actions/starter-workflows/blob/master/ci/node.js.yml

**Review by MichaelKim20 — APPROVED** [review]

File: src/Stoa.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+class Stoa {
+    public stoa: express.Application;
+
+    constructor () {
+        this.stoa = express();
+
+        this.stoa.get("/validators",
+            (req: express.Request, res: express.Response, next: express.NextFunction) => {
+            res.send(req.query.height);
```
Comment (AndrejMitrovic) [inline src/Stoa.ts:11]: Where does the `height` property come from? Is the support code for this automatically generated?

Also, height should be an optional parameter. Does this code handle that well?

File: tests/test.test.ts [hunk 2]
```diff
@@ -0,0 +1,4 @@
+describe('Calculator', () => {
```
Comment (AndrejMitrovic) [inline tests/test.test.ts:1]: Accidentally committed this?

File: tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    // "strictBindCallApply": true,           /* Enable strict 'bind', 'call', and 'apply' methods on functions. */
+    // "strictPropertyInitialization": true,  /* Enable strict checking of property initialization in classes. */
+    // "noImplicitThis": true,                /* Raise error on 'this' expressions with an implied 'any' type. */
+    // "alwaysStrict": true,                  /* Parse in strict mode and emit "use strict" for each source file. */
+
+    /* Additional Checks */
+    // "noUnusedLocals": true,                /* Report errors on unused locals. */
+    // "noUnusedParameters": true,            /* Report errors on unused parameters. */
+    // "noImplicitReturns": true,             /* Report error when not all code paths in function return a value. */
+    // "noFallthroughCasesInSwitch": true,    /* Report errors for fallthrough cases in switch statement. */
```
Comment (AndrejMitrovic) [inline tsconfig.json]: Is there a reason why these two are not enabled?

File: src/Stoa.ts (hunk 1 above)
Comment (TrustHenry) [reply src/Stoa.ts:11]: In the get method request, the query parameter is stored in a dynamic object and this is an option.
IF processing will be done in logic construction in the future.

File: tests/test.test.ts (hunk 2 above)
Comment (TrustHenry) [reply tests/test.test.ts:1]: Here's a sample of test implementation for CI.

File: src/Stoa.ts (hunk 1 above)
Comment (AndrejMitrovic) [reply src/Stoa.ts:11]: Alright!

File: tsconfig.json (hunk 3 above)
Comment (TrustHenry) [reply tsconfig.json]: Good catch! It's a good option for error detection.

File: tests/test.test.ts (hunk 2 above)
Comment (AndrejMitrovic) [reply tests/test.test.ts:1]: Aha, I see it in the CI. Alright.

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -27,4 +27,3 @@ jobs:
     - run: npm ci
     - run: npm run build --if-present
     - run: npm test
-
```
Comment (AndrejMitrovic) [inline .github/workflows/main.yml:30]: This isn't really that important, but I suggest that you and Michael have the same vs.code settings when it comes to whitespace and newlines. Otherwise whenever someone makes a PR there will always be these unrelated changes.

**Review by AndrejMitrovic — APPROVED** [review]

File: README.md [hunk 5]
```diff
@@ -1,2 +1,28 @@
 # stoa
 API server for Agora
+
+
+
+## Build instructions
+
+# Install the node.Js >= 12.18.0 (includes npm 6.14.4) (you might want to use a newer version)
```
Comment (AndrejMitrovic) [inline README.md:8]: @TrustHenry the readme says it requires node.js >= 12.18.0.

So then it's unnecessary to test node.js v10, right? We might want to disable it from the CI.

File: .github/workflows/main.yml (hunk 4 above)
Comment (TrustHenry) [reply .github/workflows/main.yml:30]: That's right. 
This is a blank mistake when I set up my initial CI.

File: README.md (hunk 5 above)
Comment (TrustHenry) [reply README.md:8]: We actually need to test the previous version.
We recommend more than the latest version here, but I would like to describe it as a guideline.
When I tried, Node 8 found a problem with the mocha test.
I want to support at least 10 version.

Comment (AndrejMitrovic) [reply README.md:8]: Oh ok, that's good!

**Review by Geod24 — APPROVED** [review]
